tests/unit/extensions/test_extensions.py

```diff
@@ -134,3 +134,102 @@ def test_project_load_extensions(fake_extensions, tmp_path):
 
     # New part
     assert parts[f"{FullExtension.NAME}/new-part"] == {"plugin": "nil", "source": None}
+
+
+def test_flask_extensions(fake_extensions, tmp_path, input_yaml, monkeypatch):
+    monkeypatch.setenv("ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS", "1")
+    extensions.register("flask", extensions.flask.Flask)
+
+    (tmp_path / "requirements.txt").write_text("flask")
+    (tmp_path / "app.py").touch()
+    (tmp_path / "static").mkdir()
+    (tmp_path / "node_modules").mkdir()
+
+    input_yaml["extensions"] = ["flask"]
+
+    applied = extensions.apply_extensions(tmp_path, input_yaml)
+
+    assert applied["run_user"] == "_daemon_"
+    assert applied["platforms"] == {"amd64": {}}
+
+    # Root snippet extends the project's
+    services = applied["services"]
+    assert services["flask"] == {
+        "command": "/bin/python3 -m gunicorn --bind 0.0.0.0:8000 app:app",
+        "override": "replace",
+        "startup": "enabled",
+        "user": "_daemon_",
+        "working-dir": "/srv/flask/app",
+    }
+
+    parts = applied["parts"]
+    assert sorted(parts["flask/install-app"]["stage"]) == [
+        "srv/flask/app/app.py",
+        "srv/flask/app/requirements.txt",
+        "srv/flask/app/static",
+    ]
+    del parts["flask/install-app"]["stage"]
```

Comment:
an alternative to sorting + deleting it here, which I assume you did because of the unpredictable ordering, is to have the extension always sort the list of stage files. Then it's more deterministic which is a plus

Reply:
Good idea, updated, thanks!